OracleSOASuite/dockerfiles/12.2.1.3/README.md

```diff
@@ -0,0 +1,170 @@
+    DOMAIN_TYPE=<soa/osb/soaosb/soaess/soaessosb/bpm>
+    
+Sample Data will look like this:
+
+    CONNECTION_STRING=soadb:1521/soapdb
+    RCUPREFIX=SOA1
+    DB_PASSWORD=Welcome1
+    DB_SCHEMA_PASSWORD=Welcome1
+    ADMIN_PASSWORD=Welcome1
+    MANAGED_SERVER=soa_server1
+    DOMAIN_TYPE=soa
+    
+To start a docker container with a SOA domain and the WebLogic Administration Server call `docker run` command and pass the above `adminserver.env.list` file.
+
+A sample docker run command is given below:
+
+     $docker run -i -t  --name soaas --network=SOANet -p 7001:7001  -v /scratch/DockerVolume/SOAVolume/SOA:/u01/oracle/user_projects   --env-file ./adminserver.env.list oracle/soa:12.2.1.3.0
+     
+>IMPORTANT: the resulting images will NOT have a domain pre-configured. But, it has the scripts to create and configure a soa domain >while creating a container out of the image.
+     
+The options `-i -t` in the above command runs the container in interactive mode and you will be able to see the commands running in the container. This includes the command for RCU creation, domain creation and configuration followed by starting the Administration Server. 
+
+>IMPORTANT: You need to wait till all the above commands are run before you can access the Administration Server Web Console.
+>The following lines highlight when the Administration Server is ready to be used:    
+
+`INFO: Admin server is running`
+
+`INFO: Admin server running, ready to start managed server`
+
+The above line indicate that the Administration Server started successfully with the name `soaas`, mapping container port 7001 to host port 7001 enables accessing of the Weblogic host outside of the local host, connecting to SOANet network enables accessing the DB container by it's name i.e soadb. This includes the command to tail logs to keep the container up and running.
+
+
+## Creating a container for SOA Server
+
+Start a container to launch the Managed Server from the image created. The environment variables used to run the Managed Server image are defined in the file `soaserver.env.list`. 
+
+Create an environment file `soaserver.env.list` with the below details:
+
+    MANAGED_SERVER=<Managed Server name, For Exp:- soa_server1, osb_server1, bpm_server1>
+    DOMAIN_TYPE=<soa/osb/bpm>
+    ADMIN_HOST=<Admin host name>
+    ADMIN_PORT=<port number where Administration Server is running>
+    ADMIN_PASSWORD=<admin_password>
+ 
+Sample Data will look like this:
+
+    MANAGED_SERVER=soa_server1
+    DOMAIN_TYPE=soa
+    ADMIN_HOST=host.domain.com
+    ADMIN_PORT=7001
+    ADMIN_PASSWORD=Welcome1
+    
+To start a docker container for SOA server you can simply call `docker run` command and passing `soaserver.env.list`. 
+
+A sample docker run command is given below:
+
+    $ docker run -i -t  --name soams --network=SOANet -p 8001:8001   --volumes-from soaas   --env-file ./soaserver.env.list oracle/soa:12.2.1.3.0 "/u01/oracle/dockertools/startMS.sh"
```

Comment:
Why does the user need to provide a command here? And why is it wrapped in quotes? The container should just know what to run, you shouldn't have to rely on a customer getting this right. Especially when it may be driven by automation or scaling.

Reply:
This is very old README which was reviewed and approved already. This hasn't been changed since then. But anyways to clarify your question, by default container starts the Admin server and then we ask user to manually start the required Managed servers. This method of startup already reviewed and approved when we released this back in 2017. We should let this continue as is and this is applicable for Docker Only environment not for orchestration using K8S. The K8S handles this as part of it's automation.